docs/sql-ref-syntax-qry-select-groupby.md

```diff
@@ -42,42 +42,60 @@ aggregate_name ( [ DISTINCT ] expression [ , ... ] ) [ FILTER ( WHERE boolean_ex
-    result values of the grouping expressions. A grouping expression may be a column alias, a column position
-    or an expression.
+    operator performs aggregation of each grouping set specified in the `GROUPING SETS` clause.
+    Similarly, `GROUP BY GROUPING SETS ((warehouse, product), (product), ())` is semantically
+    equivalent to the union of results of `GROUP BY warehouse, product`, `GROUP BY product`
+    and global aggregate.
+    
+    GROUPING SETS can be followed with normal expressions as well, such as
+    `GROUP BY GROUPING SETS (warehouse, product)`. It generates one grouping set
+    per non-empty subset of the given expressions, so `GROUP BY GROUPING SETS (warehouse, product)`
+    is equivalent to `GROUP BY GROUPING SETS ((warehouse), (product))`.
+    
+    `GROUP BY warehouse, product GROUPING SETS((warehouse, producets), (warehouse))` is semantically equivalent to
+    `GROUP BY GROUPING SETS((warehouse, produce), (warehouse))`. Under this grammar,
+    the fields appearing in `GROUPING SETS`'s grouping_set must be a subset of the columns
+    appearing in group by expression.
 
 * **ROLLUP**
 
     Specifies multiple levels of aggregations in a single statement. This clause is used to compute aggregations
     based on multiple grouping sets. `ROLLUP` is a shorthand for `GROUPING SETS`. For example,
-    `GROUP BY warehouse, product WITH ROLLUP` is equivalent to `GROUP BY GROUPING SETS
-    ((warehouse, product), (warehouse), ())`.
+    `GROUP BY warehouse, product WITH ROLLUP` or `GROUP BY ROLLUP(warehouse, product)` is equivalent to
+    `GROUP BY GROUPING SETS((warehouse, product), (warehouse), ())`.
+    `GROUP BY ROLLUP(warehouse, product, (warehouse, location))` is equivalent to
+    `GROUP BY GROUPING SETS((warehouse, product, location), (warehouse, product), (warehouse), ())`.
     The N elements of a `ROLLUP` specification results in N+1 `GROUPING SETS`.
 
 * **CUBE**
 
     `CUBE` clause is used to perform aggregations based on combination of grouping columns specified in the
     `GROUP BY` clause. `CUBE` is a shorthand for `GROUPING SETS`. For example,
-    `GROUP BY warehouse, product WITH CUBE` is equivalent to `GROUP BY GROUPING SETS
-    ((warehouse, product), (warehouse), (product), ())`.
+    `GROUP BY warehouse, product WITH CUBE` or `GROUP BY CUBE(warehouse, product)` is equivalent to 
+    `GROUP BY GROUPING SETS((warehouse, product), (warehouse), (product), ())`.
+    `GROUP BY CUBE(warehouse, product, (warehouse, location))` is equivalent to
+    `GROUP BY GROUPING SETS((warehouse, product, location), (warehouse, product), (warehouse, location),
+     (product, warehouse, location), (warehouse), (product), (warehouse, product), ())`.
     The N elements of a `CUBE` specification results in 2^N `GROUPING SETS`.
 
+* **grouping_set**
+
+    A grouping set is specified by zero or more comma-separated expressions in parentheses.
```

Comment:
let's update this. `parentheses` is not a must-have. `a` is equal to `(a)`.

Reply:
> let's update this. `parentheses` is not a must-have. `a` is equal to `(a)`.

Done